src/profile/profile.service.ts

```diff
@@ -91,12 +94,20 @@ export class ProfileService {
   }
 
   /**
-   * It deletes a profile by id
+   * It deletes a profile and its relationships  (APIs, subscriptions,AnalyticsLogs)
    * @param {string} profileId - string - The id of the profile to be deleted
    * @returns The delete method returns a DeleteResult object.
    */
   async deleteProfile(profileId: string): Promise<void> {
     try {
+      const profileAnalytics = await this.analyticsLogRepo.find({
+        where: { profileId },
+      });
+
+      if (profileAnalytics) {
```

Comment:
@johnayinde this is an array, hope this array when it empty wont make the expression a truthy/